backstory/src/test/java/com/google/sps/story/DatamuseRequestClientTest.java

```
@@ -0,0 +1,339 @@
+
+  /** a default url to test with */
+  private static final String DEFAULT_URL = "http://sample-website.com/fake?";
+
+  /** the stream handlet to use for tests in this class (necessary to mock URLConnection) */
+  private static HttpUrlStreamHandler httpUrlStreamHandler;
+ 
+  /** 
+   * Set up the URLStreamHandlerFactory with the HTTPUrlStreamHandler 
+   * in order to mock URLConnections to properly test methods 
+   * that call outside URLs. Code comes from this tutorial: 
+   * https://claritysoftware.co.uk/mocking-javas-url-with-mockito/.
+   */
+  @BeforeClass
+  public static void setUpURLStreamHandlerFactory() {
+    // Allows for mocking URL connections
+    URLStreamHandlerFactory urlStreamHandlerFactory = mock(URLStreamHandlerFactory.class);
+    URL.setURLStreamHandlerFactory(urlStreamHandlerFactory);
+ 
+    httpUrlStreamHandler = new HttpUrlStreamHandler();
+    BDDMockito.given(urlStreamHandlerFactory.createURLStreamHandler("http")).willReturn(httpUrlStreamHandler);
+  }
+
+  /**
+   * Reset the URLConnections of this stream handler before every method,
+   * so that URLConnections injected for other tests do not mess 
+   * with the current test. Code taken from this tutorial: 
+   * https://claritysoftware.co.uk/mocking-javas-url-with-mockito/.
+   */
+  @Before
+  public void reset() {
+    httpUrlStreamHandler.resetConnections();
+  }
+
+  /** 
+   * Check that IllegalArgumentException thrown for a multiple
+   * word noun.
+   */
+  @Test
+  public void multipleWordNoun() {
+    final String MULTIWORD_NOUN = "multiword noun";
+    final int DEFAULT_MAX = 5;
+
+    DatamuseRequestClient client = new DatamuseRequestClient(DEFAULT_URL);
+
+    try {
+      client.getRelatedAdjectives(MULTIWORD_NOUN, DEFAULT_MAX);
+    } catch (IllegalArgumentException exception) {
+      return; // test should pass if this exception is thrown
+    } catch (APINotAvailableException | RuntimeException exception) {
+      // we dont' want these to be thrown
```

I remember seeing this in some of your other tests:
https://www.baeldung.com/junit-assert-exception#junit-4

Is that possible here too? That would be more concise/expressive and avoid excess logic/documentation.

---

Unfortunately, I did that at first, and it was warning me I couldn't call this method without catching the exception (i.e. I couldn't just let it be thrown to test if it is thrown. As such, I adopted this alternate approach.

---

Done

---

It looks like you could instead declare that the test method throws these exceptions:
https://stackoverflow.com/a/31423712

Throwing would cause the test to fail.

---

I used the technique there to remove the catch statement for the other exceptions and kept the main catch statement related to the relevant exception so I could verify that that specific exception had been thrown. Hope this fixes it!

---

Done